test/unit/test_transaction.cpp

```diff
@@ -258,12 +260,32 @@ TEST_F(test_transaction, trivial_types_dup_iterate_by_key)
         EXPECT_CALL(api, mdb_cursor_open(test_txn, test_dbi, _))
             .WillOnce(DoAll(SetArgPointee<2>(cursor), Return(MDB_SUCCESS)));
 
+        EXPECT_CALL(
+            api,
+            mdb_cursor_get(
+                cursor,
+                Pointee(MdbValBytesAre{0x78, 0x56, 0x34, 0x12}),
+                _,
+                MDB_FIRST_DUP))
+            .WillOnce(DoAll(
+                SetArgPointee<2>(
+                    MDB_val{test_value.size(), test_value.data()}),
+                Return(MDB_SUCCESS)));
+
         EXPECT_CALL(api, mdb_cursor_close(cursor));
         EXPECT_CALL(api, mdb_txn_abort(test_txn));
     }
 
-    const auto result = transaction.iterate();
+    const auto result = transaction.iterate_by_key(0x12345678);
     ASSERT_TRUE(result);
+
+    auto const& db_view = *result;
+    auto const it = db_view.begin();
+    EXPECT_NE(it, db_view.end());
```

Comment:
Please use `ASSERT_NE` here. Iterator can not be dereferenced if the assertion fails.